includes/main.js

```diff
@@ -28,39 +28,53 @@ function initiateApp(){
 }
 function makeGallery(imageArray){
 	//use loops and jquery dom creation to make the html structure inside the #gallery section
-
 	//create a loop to go through the images in the imageArray
-		//create the elements needed for each picture, store the elements in variable
+		for (var imageArrayIndex = 0; imageArrayIndex < imageArray.length; imageArrayIndex++){
+			//$("#gallery").append(imageArray); //css class to append images
 
-		//attach a click handler to the figure you create.  call the "displayImage" function.  
-
-		//append the element to the #gallery section
-	
+		//create the elements needed for each picture, store the elements in variable
+			var fig = $('<figure>', {
+				class: "imageGallery col-xs-12 col-sm-6 col-md-4",
+				style: "background-image:url('"+ imageArray[imageArrayIndex]+"')"
+			});
+
+			var caption = $("figcaption").slice('images/');
```

Comment:
jquery slice doesn't seem to work the way you think: https://api.jquery.com/slice/

https://jsbin.com/roluwoj/edit?html,js,output

it probably interpreted 'images/' as 0, and then just gave you every element you selected.

You would want to grab the url from the array
```
var fileName = imageArray[ imageArrayIndex ];
```
then get rid of the part before the '/' by first finding it
```
var slashPosition = fileName.indexOf('/');
```
then find the dot position, again with indexOf
```
var dotPosition = fileName.indexOf('.');
```
then use slice to get everything between the two
```
var caption = fileName.slice( slashPosition, dotPosition);
```

please use this as a basis and figure out the solution.